File: Diagrammatize.Design.Pattern/code/cxx/Observer/A2/RandomNumberGenerator.cpp
#include "RandomNumberGenerator.hpp"
#include <thread>
#include <chrono>

int RandomNumberGenerator::getNumber() {
    return number;
}

void RandomNumberGenerator::execute() {
    for (int i = 0; i < 20; i++) {
        number = random.randomInt(0, 49);
        notifyObservers();
        std::this_thread::sleep_for(std::chrono::seconds(1));
    }
}

File: Diagrammatize.Design.Patterns/code/python/Observer/A2/RandomNumberGenerator.py
import random
import time

from NumberGenerator import NumberGenerator

class RandomNumberGenerator(NumberGenerator):
    def __init__(self):
        super().__init__()
        self.random = random.Random()
        self.number = 0

    def getNumber(self):
        return self.number

    def execute(self):
        for i in range(0, 20):
            self.number = self.random.randint(0, 49)
            self.notifyObservers()
            time.sleep(1)
